File: AD-HOC/1441/main.py
def max_of(h):
    max_value = h
    while h != 1:
        h = 3 * h + 1 if h % 2 else h // 2
        if h > max_value:
            max_value = h
    return max_value


while h := int(input()):
    print(max_of(h))

File: AD-HOC/1441/main.rb
def max_of(h)
  max_value = h
  while h != 1
    h = h.odd? ? 3 * h + 1 : h / 2
    max_value = h if h > max_value
  end
  max_value
end

while (h = gets.to_i).nonzero?
  puts max_of(h)
end
